common-arcana.lic

```diff
@@ -236,4 +238,56 @@ module DRCA
       return parse_mana_message(mana_msg)
     end
   end
+
+  def cast_spells(settings, spells)
+    infuse_om(!settings.osrel_no_harness, settings.osrel_amount)
+    spells.each do |name, data|
+      next if DRSpells.active_spells[name] && (data['recast'].nil? || DRSpells.active_spells[name].to_i > data['recast'])
+      while mana < 40
+        echo('Waiting on mana...')
+        pause 15
+      end
+
+      if data['ritual']
+        ritual(data)
+      else
+        cast_spell(settings, data)
+      end
+    end
+  end
+
+  def set_moon_data(data)
+    return unless data['moon']
+
+    check_moonwatch
+    # Look for a moon that will be up for the next few minutes
+    unless moon = UserVars.moons.find { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 4 }.first
+      echo "No moon available to cast #{data['abbrev']}"
+      return  # Return nil to indicate that setting moon casting data failed
+    end
+    data['cast'] = "cast #{moon}"
+    data
+  end
+
+  def check_moonwatch
+    return unless !Script.running? 'moonwatch'
+
+    echo 'moonwatch is not running. Starting it now'
+    custom_require.call('moonwatch')
+    echo "Run `#{$clean_lich_char}e autostart('moonwatch')` to avoid this in the future"
+  end
+
+  def cast_spell(settings, data)
+    return unless data = set_moon_data(data)
```

Comment:
I wanted to make the `if data['moon']` block into a guard as suggested earlier - this was the best solution i could think of. Any thoughts for a better way to handle this? The idea is to set `data['cast']` to cast on a moon if there is a visible moon, else return without casting.